crates/egui/src/text_selection/visuals.rs

```diff
@@ -73,3 +73,44 @@ pub fn paint_cursor(painter: &Painter, visuals: &Visuals, cursor_rect: Rect) {
         );
     }
 }
+
+/// Paint text cursor.
+pub fn paint_text_cursor(
+    ui: &mut Ui,
+    painter: &Painter,
+    primary_cursor_rect: Rect,
+    is_stay_cursor: bool,
+) {
+    let i_time = ui.input(|i| i.time);
```

Comment:
Currently the cursor still blinks while typing, which is quite annoying. If we instead reset a timer each time the user presses a key, it won't blink until `text_cursor_on_duration` after the user last stopped typing.

So, I think you were right to put this timer in `TextEditState`. We could put a `last_edit_time: Option<f64>` there fed by `ui.input(|i| i.time)`, and then `paint_text_cursor` is fed with `time_since_last_edit` which will be used as the basis for the animation